Delete INSFVMomentumFriction and combine with the PINSFVMomentumFriction

kyriv-lab opened the pull request:
Closes #20857

kyriv-lab committed:
Edit the PINSFVMomentumFriction to become more general Ref #20857

kyriv-lab committed:
Delete INSFVMomentumFriction and Fix discontinuous-body-forces.i Ref #20857

kyriv-lab committed:
Fix PINSFVMomentumFriction and 2d-segregated-velocity-rz-slip.i test #20857

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/2153915/) on 51e1670 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/27257/site)

This comment will be updated on new commits.

kyriv-lab committed:
Fix discontinuous-body-forces.i Ref #20857

Fix pipe_solidification.i Ref #20857

Fix NSFVBase.h and natural_circulation_digleg.i Ref #20857

Fix ins/action/errors/2d-rc-error-action.i test Ref #20857

Fix natural_circulation_pipe.i test Ref #20857

kyriv-lab committed:
Fix ins/channel-flow friction tests and regold the ones that use FORCHHEIMER #20857

kyriv-lab committed:
Fix docs and galium_melting.i example #20857

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/2160091/) on 559ab1f wanted to post the following:


## Framework coverage

Coverage did not change


## Modules coverage

### Navier stokes

<table>
  <thead>
    <tr>
      <th></th>
      <th>b02bb1</th>
      <th colspan=4>#27257 559ab1</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>84.79%</td>
      <td>84.77%</td>
      <td>-0.02%</td>
      <td>85.14%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>14612</td>
      <td>14630</td>
      <td>+18</td>
      <td>63</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>2621</td>
      <td>2628</td>
      <td>+7</td>
      <td>11</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/27257/coverage_diff/navier_stokes)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/27257/coverage/navier_stokes)

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/27257/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/27257/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/27257/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/27257/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/27257/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/27257/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/27257/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/27257/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/27257/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/27257/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/27257/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/27257/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/27257/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/27257/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/27257/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/27257/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/27257/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/27257/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/27257/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/27257/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/27257/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/27257/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/27257/coverage/scalar_transport)
- [`solid_mechanics`](https://mooseframework.inl.gov/docs/PRs/27257/coverage/solid_mechanics)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/27257/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/27257/coverage/stochastic_tools)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/27257/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/27257/coverage/xfem)

</details>

## Warnings

- `navier_stokes` new line coverage rate 85.14% is less than the suggested 90.0%



This comment will be updated on new commits.

kyriv-lab committed:
Edit test to increase coverage #20857

moosebuild commented:
Job [Framework increased sparsity](https://civet.inl.gov/job/2167858/) on ef9914d : invalidated by @grmnptr

Not related to this PR

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/2167876/) on ef9914d wanted to post the following:


The following coverage requirement(s) failed:
- Failed to generate `framework` coverage rate (required: 81.0%)

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/2167876/) on ef9914d : invalidated by @grmnptr

moosebuild commented:
All [jobs](https://civet.inl.gov/event/175129/) on ef9914d : invalidated by @grmnptr

Invalidating for coverage.

grmnptr commented on modules/navier_stokes/doc/content/source/functormaterials/INSFVMushyPorousFrictionFunctorMaterial.md:
```diff
@@ -31,10 +31,9 @@ F = C_F \frac{\rho_l}{\sqrt{K}} \,,
 where the Ergun coefficient is $C_F = 0.55$ and $\rho_l$ is the density of the liquid phase.
 
 !alert note
-This material is compatible with [INSFVMomentumFriction.md] which multiplies the
-incoming Forchheimer coefficient by the velocity magnitude; it is *incompatible*
-with [PINSFVMomentumFriction.md] which assumes the incoming Forchheimer
-coefficient already includes multiplication by the velocity magnitude.
```
I'd just say what parameters should be used in `PINSFVMomentumFriction` at this point.

kyriv1980 replied:
Done

grmnptr commented on modules/navier_stokes/examples/solidification/galium_melting.i:
```diff
@@ -180,11 +180,16 @@ Ny = 50
+    Forchheimer_name = 'Forchheimer_coeff'
+    rho = ${rho_liquid}
+    mu = ${mu}
+    use_standard = false
```
Use standard formulation would be more descriptive.

kyriv1980 replied:
renamed the flag to make it more clear.

grmnptr commented on modules/navier_stokes/include/base/NSFVBase.h:
```diff
@@ -721,6 +725,11 @@ NSFVBase<BaseType>::validParams()
       "'porous_medium_treatment' is enabled, the coefficients already contain a velocity "
       "multiplier but they are not multiplied with density yet!");
 
+  params.addParam<bool>(
+      "use_standard", true, "Flag to enable standard friction coefficient formulation");
+  params.addParam<bool>(
+      "use_superficial", true, "Flag to enable the use of superficial fluid values");
```
```suggestion
  params.addParam<bool>(
      "use_standard", true, "Flag to enable standard friction coefficient formulation");
  params.addParam<bool>(
      "use_superficial", true, "Flag to enable to incorporate porosity in the formulation.");
```

grmnptr commented on modules/navier_stokes/include/base/NSFVBase.h:
```diff
@@ -721,6 +725,11 @@ NSFVBase<BaseType>::validParams()
       "multiplier but they are not multiplied with density yet!");
 
+  params.addParam<bool>(
+      "use_standard", true, "Flag to enable standard friction coefficient formulation");
```
For the action we need more descriptive names. To make sure everybody knows this is for the friction parts.

kyriv1980 replied:
renamed the flag to make it more clear.

grmnptr reviewed (changes requested):
Nice! The design is good, just a few other user-experience related things are left.

kyriv1980 commented on modules/navier_stokes/include/base/NSFVBase.h:
```diff
@@ -1060,6 +1069,8 @@ NSFVBase<BaseType>::NSFVBase(const InputParameters & parameters)
     _energy_two_term_bc_expansion(parameters.get<bool>("energy_two_term_bc_expansion")),
     _passive_scalar_two_term_bc_expansion(
         parameters.get<bool>("passive_scalar_two_term_bc_expansion")),
+    _use_standard(parameters.get<bool>("use_standard")),
+    _use_superficial(parameters.get<bool>("use_superficial")),
```
Even if they do set them it won't produce any errors?

kyriv1980 replied:
Not setting these parameters in the action will result in error, because they will be required by the friction kernel. Setting them without the friction error will have no effect on the simulation.

grmnptr replied:
Not setting these parameters won't produce an error because there is a default associated to them. Even if you got an error it would not be useful for the user considering that it would point to objects which are under the hood and invisible to them. So I'd recommend adding the error checks at the action level.

kyriv1980 replied:
I added checks in the action for the case where we define friction. If we have friction the user needs to specify the friction formulation, and if we have porous formulation the user needs to define if they are gonna use superficial values or interstitial values.

kyriv1980 commented on modules/navier_stokes/src/fvkernels/PINSFVMomentumFriction.C:
```diff
@@ -24,10 +25,14 @@ PINSFVMomentumFriction::validParams()
-      NS::speed,
-      "The norm of the interstitial velocity. This is required for Forchheimer calculations");
+  params.addParam<MooseFunctorName>(NS::speed, "The norm of the superficial velocity.");
+  params.addParam<MooseFunctorName>("u", "The superficial velocity in the x direction.");
+  params.addParam<MooseFunctorName>("v", "The superficial velocity in the y direction.");
+  params.addParam<MooseFunctorName>("w", "The superficial velocity in the z direction.");
+  params.addParam<MooseFunctorName>(NS::porosity, 1.0, "The porosity");
```
The user is asked to provide the superficial velocity.

grmnptr replied:
I think if we are using this in a non-porous setting (so the porosity is 1), the velocity is is going to be the same as the interstitial. I don't think there is a lot of value in using superficial velocity if the superficial flag is disabled.

grmnptr replied:
If you have a setting to use superficial/nonsuperficial then for the first it is the superficial velocity and for the second it is the interstitial.

kyriv1980 replied:
For the second superficial = interstitial.

grmnptr replied:
Then rename the parameter pls.

kyriv1980 replied:
Done

kyriv1980 commented on modules/navier_stokes/src/fvkernels/PINSFVMomentumFriction.C:
```diff
@@ -24,10 +25,14 @@ PINSFVMomentumFriction::validParams()
   params.addParam<MooseFunctorName>("Darcy_name", "Name of the Darcy coefficients property.");
   params.addParam<MooseFunctorName>("Forchheimer_name",
                                     "Name of the Forchheimer coefficients property.");
+  params.addParam<bool>("use_superficial", true, "Boolean to choose the type of fluid properties.");
```
Correct... i edited the initial description to make it clear.

kyriv1980 commented on modules/navier_stokes/src/fvkernels/PINSFVMomentumFriction.C:
```diff
@@ -63,35 +71,86 @@ PINSFVMomentumFriction::PINSFVMomentumFriction(const InputParameters & params)
 // \frac{\rho}{2} F |v|
+//
+// For the non standard formulation we define :
+// \nabla p = D v + F |v| v
```
Done

kyriv-lab committed:
Address reviewer comments  #20857

Address reviewer comments v2  #20857

Address reviewer comments v3  #20857

kyriv1980 commented on modules/navier_stokes/src/fvkernels/PINSFVMomentumFriction.C:
```diff
@@ -63,35 +71,86 @@ PINSFVMomentumFriction::PINSFVMomentumFriction(const InputParameters & params)
+            "defined for this kernel and dimensions >= 3 when using the Forchheimer fromulation.");
+      superficial_velocity = ADRealVectorValue(
+          (*_u_var)(elem_arg, state), (*_v_var)(elem_arg, state), (*_w_var)(elem_arg, state));
+    }
```
Done

grmnptr commented on modules/navier_stokes/doc/content/source/fvkernels/PINSFVMomentumFriction.md:
```diff
@@ -3,32 +3,54 @@
+
+## Standard friction formulation
+
+Set parameter: "standard_friction_formulation = true"
```
Us the `[!param](parameter path)` syntax so that when people change the parameter name in the source code the documentation alerts everyone that it needs to be changed too. You can just search for the syntax above for examples.

kyriv1980 replied:
done

grmnptr commented on modules/navier_stokes/doc/content/source/fvkernels/PINSFVMomentumFriction.md:
```diff
@@ -3,32 +3,54 @@
+shown in [forchheimer-standard], [forchheimer-simple] it has a quadratic dependence on velocity.
+
+For the intersticial version of the above equations we need to define: $v_{D,i} = v_{D,i}/_epsilon$ and $rho = rho *_epsilon$
+where $\epsilon$ the porosity. To do this set parameter: "use_superficial = false"
```
```suggestion
where $\epsilon$ the porosity. To do this set parameter [!param](fvkernels/PINSFVMomentumFriction/use_superficial)  to `false`.
```

grmnptr commented on modules/navier_stokes/include/base/NSFVBase.h:
```diff
@@ -721,6 +725,11 @@ NSFVBase<BaseType>::validParams()
 
+  params.addParam<bool>(
+      "standard_friction_formulation", true, "Flag to enable standard friction formulation");
+  params.addParam<bool>(
+      "use_superficial", true, "Flag to enable the use of superficial fluid values");
```
In the action you won't need this. This will be defined by the `porous_medium_treatment=false or true`.

kyriv1980 replied:
done

grmnptr commented on modules/navier_stokes/include/base/NSFVBase.h:
```diff
@@ -1060,6 +1069,8 @@ NSFVBase<BaseType>::NSFVBase(const InputParameters & parameters)
     _passive_scalar_two_term_bc_expansion(
         parameters.get<bool>("passive_scalar_two_term_bc_expansion")),
+    _standard_friction_formulation(parameters.get<bool>("standard_friction_formulation")),
+    _use_superficial(parameters.get<bool>("use_superficial")),
```
```suggestion
```

kyriv1980 replied:
done

grmnptr commented on modules/navier_stokes/include/base/NSFVBase.h:
```diff
@@ -1931,15 +1942,29 @@ NSFVBase<BaseType>::addINSMomentumFrictionKernels()
 {
   unsigned int num_friction_blocks = _friction_blocks.size();
   unsigned int num_used_blocks = num_friction_blocks ? num_friction_blocks : 1;
+  const std::string u_names[3] = {"u", "v", "w"};
+  if (!parameters().isParamSetByUser("standard_friction_formulation"))
+  {
+    paramError("standard_friction_formulation", "User needs to define the friction formulation");
+  }
```
Move this to `checkFrictionParameterErrors` and use the `checkDependentParameterError` function. Also this error check is not correct. As we discussed before you want to error when somebody declares these parameters with no friction available. It is still unprotected with this implementation. Also, considering that you have defaults for your parameters this error check is not needed.

kyriv1980 replied:
done

grmnptr commented on modules/navier_stokes/src/fvkernels/PINSFVMomentumFriction.C:
```diff
@@ -63,35 +96,75 @@ PINSFVMomentumFriction::PINSFVMomentumFriction(const InputParameters & params)
-// solves multiply by v_D in the gatherRCData and computeSegregatedContribution methods
-// respectively, it is the job of the computeFrictionWCoefficient method to compute, for the Darcy
-// term:
+// where v denotes the either the superficial or intersticial velocity and similarly for density
```
```suggestion
// where v denotes either the superficial or interstitial velocity and similarly for density
```

grmnptr commented on modules/navier_stokes/src/fvkernels/PINSFVMomentumFriction.C:
```diff
@@ -63,35 +96,75 @@ PINSFVMomentumFriction::PINSFVMomentumFriction(const InputParameters & params)
-// respectively, it is the job of the computeFrictionWCoefficient method to compute, for the Darcy
-// term:
+// where v denotes the either the superficial or intersticial velocity and similarly for density
+// rho. Both monolithic and segregated solves get multiplied by v in the gatherRCData and
```
```suggestion
// rho. Contributions for both monolithic and segregated systems get multiplied by v in the gatherRCData and
```

grmnptr commented on modules/navier_stokes/src/fvkernels/PINSFVMomentumFriction.C:
```diff
@@ -63,35 +96,75 @@ PINSFVMomentumFriction::PINSFVMomentumFriction(const InputParameters & params)
+  else if (_use_Forchheimer_friction_model && !_speed)
+  {
+    ADRealVectorValue superficial_velocity;
+    if (_dim >= 1)
+    {
+      superficial_velocity = ADRealVectorValue((*_u_var)(elem_arg, state));
+    }
+    if (_dim >= 2)
+    {
+      superficial_velocity =
+          ADRealVectorValue((*_u_var)(elem_arg, state), (*_v_var)(elem_arg, state));
+    }
+    if (_dim >= 3)
+    {
+      superficial_velocity = ADRealVectorValue(
+          (*_u_var)(elem_arg, state), (*_v_var)(elem_arg, state), (*_w_var)(elem_arg, state));
+    }
```
```suggestion
    if (_dim >= 1)
    {
      superficial_velocity(0) =(*_u_var)(elem_arg, state);
      if (_dim >= 2)
      {
      superficial_velocity(1) =(*_v_var)(elem_arg, state);
          if (_dim >= 3)
      superficial_velocity(2) = (*_w_var)(elem_arg, state);
    }
    
    }
```

grmnptr replied:
If you have 3 dimensions you will overwrite the whole vector 3 times. It is enough to just populate the necessary components.

grmnptr commented on modules/navier_stokes/src/fvkernels/PINSFVMomentumFriction.C:
```diff
@@ -101,21 +174,27 @@ PINSFVMomentumFriction::computeSegregatedContribution()
 {
   const auto elem_arg = makeElemArg(&elem);
   const auto state = determineState();
+  auto u_functor = _u_functor(elem_arg, state);
```
```suggestion
  auto u = _u_functor(elem_arg, state);
```

kyriv1980 replied:
done

grmnptr reviewed (changes requested):
Almost there!

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2176168/) on a55a97c wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/27257/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/27257/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format f4aebfe0f810f0bb7ce645d89757a5fa04a2f339`

grmnptr commented on modules/navier_stokes/doc/content/source/fvkernels/PINSFVMomentumFriction.md:
```diff
@@ -3,32 +3,54 @@
+shown in [forchheimer-standard], [forchheimer-simple] it has a quadratic dependence on velocity.
+
+For the intersticial version of the above equations we need to define: $v_{D,i} = v_{D,i}/_epsilon$ and $rho = rho *_epsilon$
+where $\epsilon$ the porosity. To do this set parameter [!param](/FVKernels/PINSFVMomentumFriction/use_superficial)  to `false`.
```
```suggestion
where $\epsilon$ is the porosity. To do this, set parameter [!param](/FVKernels/PINSFVMomentumFriction/use_superficial)  to `false`.
```

grmnptr commented on modules/navier_stokes/include/base/NSFVBase.h:
```diff
@@ -416,6 +416,9 @@ class NSFVBase : public BaseType
   /// of the passive scalar fields
   const bool _passive_scalar_two_term_bc_expansion;
 
+  /// Flags for the friction calculation in PISFVMomentumFriction Kernel
```
```suggestion
  /// Flag for the friction calculation in PISFVMomentumFriction Kernel
```

grmnptr commented on modules/navier_stokes/include/base/NSFVBase.h:
```diff
@@ -2024,8 +2035,10 @@ NSFVBase<BaseType>::addINSMomentumFrictionKernels()
   else
   {
-    const std::string kernel_type = "INSFVMomentumFriction";
+    const std::string kernel_type = "PINSFVMomentumFriction";
```
If both of these options use the same kernel, can we somehow make this function shorter? I think there is a lot of duplication which can be deleted.

kyriv1980 replied:
Done

grmnptr reviewed (changes requested):
Almost there! Just one more thing.

grmnptr commented on modules/navier_stokes/src/fvkernels/PINSFVMomentumFriction.C:
```diff
@@ -63,35 +102,71 @@ PINSFVMomentumFriction::PINSFVMomentumFriction(const InputParameters & params)
+  ////////////////////////////////////////////////////////////////////
+  ///// Switching across formulation cases
+  ////////////////////////////////////////////////////////////////////
+  if (!_use_superficial)
+  {
+    rho = _epsilon(elem_arg, state) * _rho(elem_arg, state);
+    speed *= (1 / _epsilon(elem_arg, state));
+  }
```
I feel like the meaning of `_use_superficial` should be the opposite. If we use it we are providing superficial quantities so we need to have these corrections. Otherwise no corrections are needed.

grmnptr replied:
Also speed is already interstitial in the material so if `_speed` is provided then you don't need to do anything here.

kyriv1980 replied:
fixed it

grmnptr reviewed (changes requested):
I apologize for these, I just notices something.

kyriv1980 commented on modules/navier_stokes/include/base/NSFVBase.h:
```diff
@@ -1931,55 +1938,67 @@ NSFVBase<BaseType>::addINSMomentumFrictionKernels()
-    const std::string kernel_type = "PINSFVMomentumFriction";
-    InputParameters params = getFactory().getValidParams(kernel_type);
-    params.template set<MooseFunctorName>(NS::density) = _density_name;
+    params.template set<bool>("use_superficial") = false;
```
when we have a porous formulation we use intestitial quanties, so divide by epsilon where applicable.

kyriv1980 commented on modules/navier_stokes/src/fvkernels/PINSFVMomentumFriction.C:
```diff
@@ -24,10 +25,22 @@ PINSFVMomentumFriction::validParams()
-  params.addParam<MooseFunctorName>(
-      NS::speed,
-      "The norm of the interstitial velocity. This is required for Forchheimer calculations");
+  params.addParam<MooseFunctorName>(NS::speed, "The norm of the superficial velocity.");
```
Are you sure?

kyriv1980 replied:
fixed it.

kyriv1980 commented on modules/navier_stokes/src/fvkernels/PINSFVMomentumFriction.C:
```diff
@@ -24,10 +25,22 @@ PINSFVMomentumFriction::validParams()
-      NS::speed,
-      "The norm of the interstitial velocity. This is required for Forchheimer calculations");
+  params.addParam<MooseFunctorName>(NS::speed, "The norm of the superficial velocity.");
+  params.addParam<MooseFunctorName>("u",
+                                    "The velocity in the x direction. Superficial in the case of "
+                                    "porous treatment, interstitial otherwise");
```
fixed it

grmnptr reviewed (changes requested):
One more line! I just want to make sure everything is consistent between objects.

grmnptr reviewed (changes requested):
Sorry, forgot to add it to the review.

kyriv1980 commented on modules/navier_stokes/src/fvkernels/PINSFVMomentumFriction.C:
```diff
@@ -63,35 +101,72 @@ PINSFVMomentumFriction::PINSFVMomentumFriction(const InputParameters & params)
+  if (_is_porous_medium)
+  {
+    rho = _epsilon(elem_arg, state) * _rho(elem_arg, state);
+    speed *= (1 / _epsilon(elem_arg, state));
```
Done

grmnptr reviewed (approved):
Looks good to me! Thanks for the contribution! Just a note for others: three tests are regolded because they relied on the C*|vel_compinent|*vel_component instead of C*|speed|*vel_component.

moosebuild commented:
Job [Apptainer moose](https://civet.inl.gov/job/2191928/) on cd9e7a1 : invalidated by @GiudGiud

unrelated?

moosebuild commented:
Job [Internal app tests](https://civet.inl.gov/job/2191952/) on cd9e7a1 : invalidated by @GiudGiud

Pgh patched

kyriv1980 commented:
@GiudGiud I saw you merged the Pronghorn patch related to this PR. Are you gonna merge this?

GiudGiud commented:
Probably should. Though do we want to wait on the NRC training on that one too @grmnptr

GiudGiud commented:
I'll rebase this. 
It will be easier than it would have been to rebase my PR

GiudGiud committed:
Remove INSFVmomentumFriction from physics
Add default empty friction back to NSFVBase

GiudGiud commented:
ok this should be rebased. I ll watch the test to make sure we pass everything.

kyriv1980 commented:
@GiudGiud  Can you force push your changes again please?

GiudGiud reviewed (approved):
thanks for the PR good cleanup

grmnptr commented:
I suppose it won't matter considering that they will use BC which was built last week, but yeah we agreed not to merge this until the training was done.

GiudGiud commented:
So when we did this we changed from the quadratic term being F u^2 to F u |u v w|

This gave diffs on the VTB, which was expected, but convergence seems slightly worse too, which is undesirable.
I m thinking we want to have an option to go back to the old version?

kyriv1980 commented:
Won't that get too complicated? It already has 4 flavors as is... which is a bit much in my opinion.

GiudGiud commented:
I can go either way. I like the robustness personally, and the solutions are very close

@tanoret @grmnptr

grmnptr commented:
Well, if you use the whole velocity vector you will have additional, norm-based coupling between the different velocity blocks which can be a potential reason for slower convergence but I would be surprised if it resulted in failed convergence. Which model exhibits this issue? How much worse is the convergence? If it is not too much worse, I would vote for keeping the norm of the velocity there.

GiudGiud commented:
MSFR VTB input
https://github.com/idaholab/virtual_test_bed/blob/devel/msr/msfr/transient/run_ns.i
